### PR TITLE
[None][feat] Add processed logprobs

### PR DESCRIPTION
**Pull request by dominicshanshan**
## Release Notes

* **New Features**
  * Introduced `logprobs_mode` parameter to configure log probability computation
  * Log probabilities now apply sampling transformations (temperature, top-k, top-p filtering) to logits before processing
  * New "processed_logprobs" mode enabled by default with full backward compatibility

* **Tests**
  * Added comprehensive test coverage for processed logprobs across various sampling strategies and temperature settings

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--reuse-test (optional)pipeline-id --disable-fail-fast --skip-test --stage-list "A10-PyTorch-1, xxx" --gpu-type "A30, H100_PCIe" --test-backend "pytorch, cpp" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx" --detailed-log --debug(experimental)]`

Launch build/test pipelines. All previously running jobs will be killed.

`--reuse-test (optional)pipeline-id ` *(OPTIONAL)* : Allow the new pipeline to reuse build artifacts and skip successful test stages from a specified pipeline or the last pipeline if no pipeline-id is indicated. If the Git commit ID has changed, this option will be always ignored. The DEFAULT behavior of the bot is to reuse build artifacts and successful test results from the last pipeline.

`--disable-reuse-test ` *(OPTIONAL)* : Explicitly prevent the pipeline from reusing build artifacts and skipping successful test stages from a previous pipeline. Ensure that all builds and tests are run regardless of previous successes.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-PyTorch-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-PyTorch-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--test-backend "pytorch, cpp"` *(OPTIONAL)* : Skip test stages which don't match the specified backends. Only support [pytorch, cpp, tensorrt, triton]. Examples: "pytorch, cpp" (does not run test stages with tensorrt or triton backend). Note: Does **NOT** update GitHub pipeline status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests in addition to running L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx".

`--detailed-log ` *(OPTIONAL)* : Enable flushing out all logs to the Jenkins console. This will significantly increase the log volume and may slow down the job.

`--debug ` *(OPTIONAL)* : **Experimental feature**. Enable access to the CI container for debugging purpose. Note: Specify exactly one stage in the `stage-list` parameter to access the appropriate container environment. Note: Does **NOT** update GitHub check status.

For guidance on mapping tests to stage names, see `docs/source/reference/ci-overview.md`
and the `scripts/test_to_stage_mapping.py` helper.

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>

### CODE REVIEW

**Comment by dominicshanshan** [comment]
`logprobs` API refactor and make the `logprobs_mode` with multiple return will filed in separate PR, this PR only implement `processed_logprobs` for RL usage.

File: tests/unittest/llmapi/test_llm_pytorch.py [hunk 1]
```diff
@@ -915,6 +915,214 @@ def test_llm_return_logprobs_streaming(prompt_logprobs, logprobs,
...
+        assert val <= 0.0, f"processed_logprobs should have non-positive values, got {val}"
+
+    del llm
+
+
+@skip_ray
+@pytest.mark.parametrize("temperature", [0.5, 1.0, 1.5])
+def test_llm_processed_logprobs_with_temperature(temperature):
+    """
+    Test that processed_logprobs correctly applies temperature scaling.
```
Comment (stnie) [inline tests/unittest/llmapi/test_llm_pytorch.py]: I think `test_llm_logprobs_modes_basic` tests the same as this test

File: tests/unittest/llmapi/test_llm_pytorch.py [hunk 2]
```diff
@@ -915,6 +915,214 @@ def test_llm_return_logprobs_streaming(prompt_logprobs, logprobs,
...
+    outputs = list(llm.generate(prompt, params))
+    assert len(outputs) == 1
+
+    # Check that some logprobs are -inf (masked by top-p) across all generated tokens
+    # Note: With top_p, not every token position will have -inf values in the top-k logprobs
+    # We need to check across all tokens.
+    all_logprobs = outputs[0].outputs[0].logprobs
+    for token_idx, token_logprobs in enumerate(all_logprobs):
+        logprob_values = [obj.logprob for obj in token_logprobs.values()]
+        if token_idx == 0:
```
Comment (stnie) [inline tests/unittest/llmapi/test_llm_pytorch.py]: I think this does not check all returned logprobs, if `-inf` is encountered early. 
I also think `float("-inf") < 0` should be True, so this may not be needed at all.

File: tensorrt_llm/sampling_params.py [hunk 3]
```diff
@@ -174,6 +181,9 @@ class SamplingParams:
 
         logprobs (int, optional): Number of log probabilities to return per output token. Defaults to None.
         prompt_logprobs (int, optional): Number of log probabilities to return per prompt token. Defaults to None.
+        logprobs_mode (str): Controls return logprobs after sampling modifications. Defaults to "processed_logprobs".
+            Options:
+            - "processed_logprobs": Return log-softmax of processed logits
```
[the remark this answers is not in the document]
Comment (dominicshanshan) [reply tensorrt_llm/sampling_params.py]: Need to define `raw_logprobs` before implement, there is already `logprobs` which do raw_logits -> log_softmax() -> top_k() in `SamplingParams`

File: tensorrt_llm/_torch/pyexecutor/sampler.py [hunk 4]
```diff
@@ -1747,10 +1757,36 @@ def _process_requests(
...
+            if logprobs_mode == "processed_logprobs":
+                # Process logits with the same transformations as sampling (temperature, top-k, top-p)
+                # but without actually sampling
+                processed_logits_list = []
+                for req_id in logprobs_req_indices:
+                    req = requests[req_id]
+                    strategy = _request_strategy(req, vocab_size=logits_cuda.size(1))
+                    req_logits_indices = logits_cuda_indexer[req_id]
+                    req_logits = logits_cuda[req_logits_indices].to(
+                        dtype=torch.float32, non_blocking=True
```
[the remark this answers is not in the document]
Comment (dominicshanshan) [reply tensorrt_llm/_torch/pyexecutor/sampler.py]: This belong to optimization modification which to use per batch process instead per `request_id`, let me make this happen in my second PR which focus on `logprobs` API re-organize.

File: tensorrt_llm/_torch/pyexecutor/llm_request.py [hunk 5]
```diff
@@ -778,6 +778,8 @@ def executor_request_to_llm_request(
         py_multimodal_data=getattr(executor_request, "py_multimodal_data",
                                    None),
         kv_cache_retention_config=executor_request.kv_cache_retention_config)
+    if hasattr(executor_request, "_logprob_params"):
+        llm_request._logprob_params = executor_request._logprob_params
```
[the remark this answers is not in the document]
Comment (dominicshanshan) [reply tensorrt_llm/_torch/pyexecutor/llm_request.py]: agree

File: tensorrt_llm/_torch/pyexecutor/sampler.py [hunk 6]
```diff
@@ -1747,10 +1757,36 @@ def _process_requests(
...
+                    )
+                    # Apply the same processing as sampling would apply
+                    processed_req_logits = process_logits(strategy, req_logits)
```
[the remark this answers is not in the document]
Comment (dominicshanshan) [reply tensorrt_llm/_torch/pyexecutor/sampler.py]: add this change in second PR

File: tensorrt_llm/sampling_params.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (dominicshanshan) [reply tensorrt_llm/sampling_params.py]: Plan in the separate PR to add `raw_logprobs` as option in `logprobs_mode`, if this `raw_logprobs` can replace current `logprobs` is debatable, however.
